pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -260,77 +262,83 @@ func (r *ReconcileClusterDeployment) Reconcile(request reconcile.Request) (recon
 		return reconcile.Result{}, err
 	}
 
-	cdLog.Debug("loading pull secret secret")
-	pullSecret, err := r.loadSecretData(cd.Spec.PullSecret.Name, cd.Namespace, pullSecretKey)
-	if err != nil {
-		cdLog.WithError(err).Error("unable to load pull secret from secret")
-		return reconcile.Result{}, err
-	}
-
 	if cd.Status.InstallerImage == nil {
 		return reconcile.Result{}, r.resolveInstallerImage(cd, hiveImage, cdLog)
 	}
 
-	job, cfgMap, err := install.GenerateInstallerJob(
-		cd,
-		hiveImage,
-		releaseImage,
-		serviceAccountName,
-		sshKey,
-		pullSecret)
-	if err != nil {
-		cdLog.WithError(err).Error("error generating install job")
-		return reconcile.Result{}, err
-	}
-
-	if err = controllerutil.SetControllerReference(cd, job, r.scheme); err != nil {
-		cdLog.WithError(err).Error("error setting controller reference on job")
-		return reconcile.Result{}, err
-	}
-	if err = controllerutil.SetControllerReference(cd, cfgMap, r.scheme); err != nil {
-		cdLog.WithError(err).Error("error setting controller reference on config map")
+	// Check if an install job already exists:
+	existingJob := &batchv1.Job{}
+	installJobName := install.GetInstallJobName(cd)
+	err = r.Get(context.TODO(), types.NamespacedName{Name: installJobName, Namespace: cd.Namespace}, existingJob)
+	if err != nil && errors.IsNotFound(err) {
+		cdLog.Debug("no install job exists")
+		existingJob = nil
+	} else if err != nil {
+		cdLog.WithError(err).Error("error looking for install job")
 		return reconcile.Result{}, err
 	}
 
-	cdLog = cdLog.WithField("job", job.Name)
+	if cd.Status.Installed {
```

Comment:
should this check happen before we try to check for the existence of a job?

Reply:
existingJob is used regardless when we calculate status down below, so this part was still required, but we could do further refactoring if you would prefer.

Reply:
No, I think it's fine as it is... missed the part below. Thanks.